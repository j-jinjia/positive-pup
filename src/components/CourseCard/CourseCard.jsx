import "./CourseCard.scss";
import dogimg from "../../assets/images/dogimg-removebg-preview.png";
const CourseCard = () => {
  return (
    <>
      <div className="course-card">
        <img className="course-card__img" src={dogimg} alt="dog image" />
        <div className="course-content">
          <h5 className="course-content__type"> Group Class</h5>
          <h3 className="course-content__heading"> Little Rising Stars </h3>
        </div>
      </div>
    </>
  );
};

export default CourseCard;
