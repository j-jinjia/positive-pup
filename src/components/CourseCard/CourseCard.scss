@use "../../styles/variables" as *;

.course-card {
  background-color: $colour-secondary-background;
  max-width: 23.125rem;
  max-height: 24.125rem;
  filter: drop-shadow(0px 20px 40px rgba(0, 0, 0, 0.04));
  border-radius: 0.5rem;
  display: grid;
  grid-template-rows: 2fr 1fr;

  .course-card__img-container {
    overflow: hidden;

    .course-card__img {
      height: 100%;
      padding: 0 3rem;
    }
  }

  &__course-content {
    background-color: $colour-secondary;
    padding: 1rem;
    border-radius: 0 0 0.5rem 0.5rem;

    .course-card__course-type {
      color: $colour-primary;
    }

    .course-card__course-heading {
      margin-bottom: 1rem;
    }
  }

  &__labels {
    display: flex;
    flex-direction: row;
    gap: 1rem;
  }
}
