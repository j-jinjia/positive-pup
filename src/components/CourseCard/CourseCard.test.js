import { customRender } from "../../utils/testUtils";
import CourseCard from "./CourseCard";
import dogImage from "../../assets/images/dog-image.png";

it("should match CourseCard component snapshot", () => {
  const { container } = customRender(
    <CourseCard
      image={dogImage}
      courseType="Group Class"
      courseHeading="Little Stars"
      labelsArray={["week", "online"]}
    />
  );

  expect(container).toMatchSnapshot();
});
