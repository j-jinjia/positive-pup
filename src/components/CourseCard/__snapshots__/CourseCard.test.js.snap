// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match CourseCard component snapshot 1`] = `
<div>
  <div
    class="course-card"
  >
    <div
      class="course-card__img-container"
    >
      <img
        alt="dog image"
        class="course-card__img"
        src="dog-image.png"
      />
    </div>
    <div
      class="course-card__course-content"
    >
      <p
        class="course-card__course-type"
      >
        Group Class
      </p>
      <p
        class="course-card__course-heading"
      >
        Little Stars
      </p>
      <div
        class="course-card__labels"
      >
        <div
          class="card-label"
        >
          <span
            class="card-label__text"
          >
            week
          </span>
        </div>
        <div
          class="card-label"
        >
          <span
            class="card-label__text"
          >
            online
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
